lib/check/data_sources.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "csv"
+require "mods"
+require File.expand_path("../../errors/missing_data.rb", __FILE__)
+require File.expand_path("../../errors/wrong_encoding.rb", __FILE__)
+require File.expand_path("../../util.rb", __FILE__)
+
+module Check
+  # Checks that the files specified in metadata actually exist
+  module DataSources
+    DATA_ROOT = Pathname.new("/opt/ingest").freeze
+
+    # @param [Array<String>] files
+    # @return [Array<MetadataError>]
+    def self.batch(files)
+      unless Dir.exist? DATA_ROOT
+        $stderr.puts "#{Util.warn("WARNING:")} "\
+                     "#{DATA_ROOT} does not exist; "\
+                     "skipping checks."
+        return []
+      end
+
+      files.map do |file|
+        paths_for(file).map do |path|
+          next if DATA_ROOT.join(path).exist?
```

Comment:
another use case to either add to this or as a separate check just came up here: ensuring the files (assuming they do exist on disk) aren't empty. so something like a `File.zero? DATA_ROOT.join(path)`-ish

Should this ever happen? No. Has is happened at UCSD? Yes.

Reply:
🙀